### PR TITLE
Deprecate old or unused Model methods

### LINKED ISSUE #5859: Remove `t` suffix from `Model` methods
`model.logpt`, `model.dlogpt` and `model.d2logpt` would be a bit more intuitive without the `t` suffix. Historically that was done to distinguish the `tensor` graphs from the compiled functions, but those are now obtained via `model.compile_logp`, `model.compile_dlogp` and `model.compile_d2logp`.

### PR DESCRIPTION
**Pull request by ricardoV94**
Closes #5859 

## Major / Breaking Changes
- Deprecate several methods from `Model` class

## Bugfixes / New features
- ...

## Docs / Maintenance
- ...

### CODE REVIEW

File: pymc/model.py [hunk 1]
```diff
@@ -983,6 +907,32 @@ def unobserved_value_vars(self):
...
+
+    @property
+    def discrete_value_vars(self):
+        """All the discrete value variables in the model"""
+        return list(typefilter(self.value_vars, discrete_types))
+
+    @property
+    def cont_vars(self):
+        warnings.warn(
+            "Model.continuous_value_vars has been deprecated. Use Model.continuous_value_vars instead.",
```
Comment (twiecki) [inline pymc/model.py]:
```suggestion
            "Model.cont_vars has been deprecated. Use Model.continuous_value_vars instead.",
```

File: pymc/model.py [hunk 2]
```diff
@@ -983,6 +907,32 @@ def unobserved_value_vars(self):
 
         return vars + untransformed_vars + deterministics
 
+    @property
+    def disc_vars(self):
```
Comment (twiecki) [inline pymc/model.py:911]: Has this been here before?

File: pymc/model.py (hunk 1 above)
Comment (ricardoV94) [reply pymc/model.py]: Good catch

File: pymc/model.py (hunk 2 above)
Comment (ricardoV94) [reply pymc/model.py:911]: It was not here before, but existed already. I tried to regroup methods to follow a more logical order but it looks like an addition in this specific diff :P

**Review by michaelosthege — APPROVED** [review]
Flaky tests being flaky again.

LGTM

**Review by twiecki — APPROVED** [review]

**Pull request merged**